packages/wrangler/src/inspect.ts

```diff
@@ -0,0 +1,526 @@
+ */
+function logConsoleMessage(evt: Protocol.Runtime.ConsoleAPICalledEvent): void {
+  const args = [];
+  for (const ro of evt.args) {
+    switch (ro.type) {
+      case "string":
+      case "number":
+      case "boolean":
+      case "undefined":
+      case "symbol":
+      case "bigint":
+        args.push(ro.value);
+        break;
+      case "function":
+        args.push(`[Function: ${ro.description}]`);
+        break;
+      case "object":
+        if (!ro.preview) {
+          args.push(ro.description);
+        } else {
+          args.push(ro.preview.description);
+
+          switch (ro.preview.subtype) {
+            case "array":
+              args.push(
+                "[ " +
+                  ro.preview.properties
+                    .map(({ value }) => {
+                      return value;
+                    })
+                    .join(", ") +
+                  (ro.preview.overflow ? "..." : "") +
+                  " ]"
+              );
+
+              break;
+            case "map":
+              args.push(
+                "{\n" +
+                  ro.preview.entries
+                    .map(({ key, value }) => {
+                      return `  ${key.description} => ${value.description}`;
+                    })
+                    .join(",\n") +
+                  (ro.preview.overflow ? "\n  ..." : "") +
+                  "\n}"
+              );
+
+              break;
+            case "set":
+              args.push(
+                "{ " +
+                  ro.preview.entries
+                    .map(({ value }) => {
+                      return `${value.description}`;
+                    })
+                    .join(", ") +
+                  (ro.preview.overflow ? ", ..." : "") +
+                  " }"
+              );
+
+              break;
+            case "null":
+              args.push("null");
+              break;
+            case "node":
+            case "regexp":
+            case "date":
+            case "weakmap":
+            case "weakset":
+            case "iterator":
+            case "generator":
+            case "error":
+            case "proxy":
+            case "promise":
+            case "typedarray":
+            case "arraybuffer":
+            case "dataview":
+            case "webassemblymemory":
+            case "wasmvalue":
+              break;
+            default:
+              // just a pojo
+              args.push(
+                "{\n" +
+                  ro.preview.properties
+                    .map(({ name, value }) => {
+                      return `  ${name}: ${value}`;
+                    })
+                    .join(",\n") +
+                  (ro.preview.overflow ? "\n  ..." : "") +
+                  "\n}"
+              );
+          }
+        }
+        break;
+      default:
+        args.push(ro.description || ro.unserializableValue || "🦋");
+        break;
+    }
+  }
+
+  console[evt.type](...args);
```

Comment:
Given that this is running in ink, I wonder if we could pass these messages out to be rendered in an Ink component, rather than just dumping them out using `console.xxx`?
(I appreciate that Ink [patches `console`](https://github.com/vadimdemedes/ink#patchconsole) but I think we could be more elegant here...

Reply:
We can revisit it. I actually try to avoid Ink for anything that doesn't feel like UI (and before you say it, I understand the boudnary is deeply fuzzy.) For example, in this usecase, I want the messages to persist no matter what, even if the process exits, etc. For now I think calling `console` methods makes sense, and matches behaviour closer to node (or at least, is already better than what we have, which is nothing.)